lib/generators/rspec/integration/templates/api_request_spec.rb

```diff
@@ -0,0 +1,154 @@
+require 'rails_helper'
+
+# This spec was generated by rspec-rails when you ran the scaffold generator.
+# It demonstrates how one might use RSpec to specify the controller code that
+# was generated by Rails when you ran the scaffold generator.
+#
+# It assumes that the implementation code is generated by the rails scaffold
+# generator.  If you are using any extension libraries to generate different
+# controller code, this generated spec may or may not pass.
+#
+# It only uses APIs available in rails and/or rspec-rails.  There are a number
+# of tools you can use to make these specs even more expressive, but we're
+# sticking to rails and rspec-rails APIs to keep things simple and stable.
+
+
+<% module_namespacing do -%>
+RSpec.describe <%= class_name.pluralize %>Controller, <%= type_metatag(:request) %> do
+
+  # This should return the minimal set of attributes required to create a valid
+  # <%= class_name %>. As you add validations to <%= class_name %>, be sure to
+  # adjust the attributes here as well.
+  let(:valid_attributes) {
+<% if options[:factory] || Gem.loaded_specs.has_key?('factory_bot_rails') -%>
+     attributes_for(:<%= ns_file_name -%>)
+<% elsif options[:fabrication] || Gem.loaded_specs.has_key?('fabrication')-%>
+    Fabricate.attributes_for(<%= ns_prefix.empty? ? ":"+ns_file_name : "'#{class_name}'"-%>)
+<% else -%>
+    skip("Add a hash of attributes valid for your model")
+<% end -%>
+  }
+
+  let(:invalid_attributes) {
+    skip("Add a hash of attributes invalid for your model")
+  }
+
+  # This should return the minimal set of values that should be in the headers
+  # in order to pass any filters (e.g. authentication) defined in
+  # <%= class_name %>Controller. Be sure to keep this updated too.
+  # Because it's an API request a default JSON header is added.
+  # If you have an Authentication - and you definitely should - add your token header here.
+  let(:valid_headers) {
+    {
+      "Content-Type" => "application/json"
+    }
+  }
+
+<% unless options[:singleton] -%>
+  describe "GET #index" do
+    it "returns a success response" do
+      <%= file_name %> = <%= class_name %>.create! valid_attributes
+
+      get <%= ns_file_name.pluralize %>_path, params: {}, headers: valid_headers
```

Comment:
I dont understand what you mean... :-/

Reply:
Older versions of rails accepted a hash of options to their `get/post/put` etc methods, some time around Rails 4 this was changed to keyword arguments. If these are keyword arguments this is fine, but if this is a hash of arguments we tend to favour the hashrocket `:key => value` style due to older Ruby support. Did you copy these from the controller equivalent? If so thats fine as this will be the same level of support.

Reply:
This needs resolving

Reply:
This can now safely be ignored, you can use rails 5 syntax.